lib/FusionInventory/Agent/Inventory.pm

```diff
@@ -250,7 +250,26 @@ sub addEntry {
         $entry->{SERIALNUMBER} = $entry->{SERIAL} if !$entry->{SERIALNUMBER}
     }
 
-    push @{$self->{content}{$section}}, $entry;
+    if ($params{identity} && ref $params{identity} eq 'ARRAY') {
```

Comment:
it's not just a grep. It's one grep, i look if there's one only retrieved. If so, it's done and i take this element. If not, i try an other grep. It van't be done in one grep.